account-info/src/lib.rs

```diff
@@ -164,26 +204,46 @@ pub fn install_or_upgrade_contract(name: String) {
 /// Needs to start with `http://` or `https://`.
 #[no_mangle]
 fn set_url() {
+    let caller = utils::get_caller();
     let url: String = runtime::get_named_arg("url");
-
-    if !check_url(&url) {
-        revert(ContractError::BadUrlFormat)
-    }
-    set_key(&runtime::get_caller().to_string(), url);
+    let purse: Option<URef> = runtime::get_named_arg("purse");
+    Deposits::new().deposit_if_needed(&caller, purse);
+    Urls::new().set(&caller, &url);
```

Comment:
It's not clear if anyone outside of the contract supposed to call this entry point. Can it be merged with ```call_set_url```? 

Additionally, the ```purse``` argument is not defined in ```get_entry_points```. That confused me.